compliance/azure/ahub_manual/Azure_AHUB_Utilization_with_Manual_Entry.pt

```diff
@@ -0,0 +1,298 @@
+          result:result,
+          summary_template: total_vms_with_ahub+" "+vm+" with AHUB enabled whose sum of cores is "+total_cores_with_ahub
+	    };
+      }else if(total_cores_with_ahub < license_cores && total_vms_with_ahub > license_ahub){
+        final_result={note:"It is observed that a greater number of licenses have been consumed than allocated licenses, Please disable AHUB for any of the "+(total_vms_with_ahub - license_ahub)+" VMs listed above.",
+          result:result,
+          summary_template: total_vms_with_ahub+" "+vm+" with AHUB enabled whose sum of cores is "+total_cores_with_ahub
+	    };
+	  }else if(total_cores_with_ahub > license_cores && total_vms_with_ahub < license_ahub){
+        final_result={note:"It is observed that a greater number of licenses have been consumed than allocated licenses, Please disable AHUB for VMs listed above whose sum of cores should be minimum "+(total_cores_with_ahub - license_cores)+" cores.",
+          result:result,
+          summary_template: total_vms_with_ahub+" "+vm+" with AHUB enabled whose sum of cores is "+total_cores_with_ahub
+	    };
+	  }
+    }else if(total_cores_with_ahub < license_cores && total_vms_with_ahub < license_ahub){
+      result=without_ahub;
+	  if(total_vms_without_ahub == 1){
+        vm = "VM";
+	  }
+      final_result={note:"It is observed that a lesser number of licenses have been consumed than allocated licenses, You can enable AHUB upto "+(license_ahub - total_vms_with_ahub)+" VMs whose sum of cores should be lesser or equal to "+(license_cores - total_cores_with_ahub)+" cores.",
+        result:result,
+        summary_template: total_vms_without_ahub+" "+vm+" with AHUB disabled"
+	  };
+    }else{
+      final_result={result:result};
+    }
+  EOS
+end
+
+script "js_results_size", type: "javascript" do
+  parameters "final_result"
+  result "result"
+  code <<-EOS
+    var result = final_result.result;
+  EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy 'azure_ahub_policy' do
+  validate $ds_vminstance_with_cores do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{data.summary_template}}"
+    detail_template <<-EOS
+# Azure AHUB Utilization with Manual Entry
+| Instance Name | Resource Group | Location | Instance Size | Image Publisher | Image Offer | Image Sku | Image Version |  Tags | Number Cores |
+| ------------- | -------------- | -------- | ------------- | --------------- | ----------- | --------- | ------------- |  ---- | ------------ |
+{{ range data.result -}}
+| {{.name}} | {{.resourceGroup}} | {{.location}} | {{.vmSize}} | {{.imagePublisher}} | {{.imageOffer}} | {{.imageSku}} | {{.imageVersion}} | {{.tags}} | {{.numberofcores}} |
+{{ end -}}
+#### Note: {{data.note}}
```

Comment:
This isn't formatting correctly on the incident page

Reply:
Thanks, we made the required changes.